File: download/upload_trigenclaw.py
"""上传 TriGenClaw 桌面端安装包到服务器"""
import paramiko
import os
import time

HOST = '120.79.17.184'
USER = 'admin'
PASS = 'Wangjie910621'
PORT = 22

local_file = r'G:\大模型聚合网站\download\trigenclaw-build\release\TriGenClaw-1.0.0-win-Setup.exe'
remote_dir = '/home/admin/ai-nexus/download'

print(f'安装包: {local_file}')
print(f'大小: {os.path.getsize(local_file) / 1024 / 1024:.1f} MB')

ssh = paramiko.SSHClient()
ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())

try:
    ssh.connect(HOST, port=PORT, username=USER, password=PASS, look_for_keys=False, allow_agent=False, timeout=30)
    print(f'已连接到 {HOST}')
    
    # 创建远程目录
    _, stdout, stderr = ssh.exec_command(f'mkdir -p {remote_dir}')
    stdout.channel.recv_exit_status()
    
    # 上传文件
    sftp = ssh.open_sftp()
    remote_file = f'{remote_dir}/TriGenClaw-1.0.0-win-Setup.exe'
    
    # 使用 paramiko 的 put 方法
    def progress_callback(transferred, total):
        pct = transferred / total * 100
        print(f'\r上传进度: {pct:.1f}% ({transferred/1024/1024:.1f}MB)', end='')
    
    sftp.put(local_file, remote_file, callback=progress_callback)
    print(f'\n上传完成: {remote_file}')
    sftp.close()
    
    # 设置权限
    ssh.exec_command(f'chmod 644 {remote_file}')
    
    # 检查下载目录
    _, stdout, _ = ssh.exec_command(f'ls -la {remote_dir}/')
    print(f'\n下载目录:\n{stdout.read().decode()}')
    
    # 检查现有的下载链接
    _, stdout, _ = ssh.exec_command(f'grep -n "download" /home/admin/ai-nexus/index.html 2>/dev/null | head -20')
    print(f'当前下载链接:\n{stdout.read().decode()}')
    
    print('\n✅ 上传完成!')
    
except paramiko.AuthenticationException:
    print(f'认证失败 - 密码可能已更改')
    # 尝试其他密码或方法
except Exception as e:
    print(f'错误: {e}')
finally:
    ssh.close()

File: download/upload-trigenclaw.js
/* ========================================
   上传 TriGenClaw 桌面端安装包到服务器
   并更新下载页面
   ======================================== */
const { Client } = require('ssh2');
const path = require('path');
const fs = require('fs');

const HOST = '120.79.17.184';
const USER = 'admin';
const { DEPLOY_PASS } = require('./deploy-config');
const PASS = DEPLOY_PASS;
const KEY_FILE = 'C:\\Users\\WBS—JCNC\\.ssh\\id_ed25519';
const INSTALLER = path.join(__dirname, 'trigenclaw-build', 'release', 'TriGenClaw-1.0.0-win-Setup.exe');
const REMOTE_DIR = '/home/admin/ai-nexus';

if (!fs.existsSync(INSTALLER)) {
  console.error('错误: 找不到安装包文件:', INSTALLER);
  process.exit(1);
}

console.log('安装包大小:', (fs.statSync(INSTALLER).size / 1024 / 1024).toFixed(1), 'MB');

const conn = new Client();

conn.on('ready', () => {
  console.log('已连接到', HOST);

  // 创建 download 目录
  conn.exec('mkdir -p ' + REMOTE_DIR + '/download', (err) => {
    if (err) console.log('mkdir 警告:', err.message);

    conn.sftp((err, sftp) => {
      if (err) { console.error('SFTP 错误:', err); conn.end(); return; }

      // 上传安装包
      const remoteFile = REMOTE_DIR + '/download/TriGenClaw-1.0.0-win-Setup.exe';
      const readStream = fs.createReadStream(INSTALLER);
      const writeStream = sftp.createWriteStream(remoteFile);

      writeStream.on('close', () => {
        console.log('安装包上传完成:', remoteFile);

        // 获取文件权限
        conn.exec('chmod 644 ' + remoteFile, (err) => {
          if (err) console.log('chmod 警告:', err.message);
        });

        // 检查服务器上的下载页面
        conn.exec('ls -la ' + REMOTE_DIR + '/download/', (err, stream) => {
          if (err) { console.error('ls 错误:', err); conn.end(); return; }
          let output = '';
          stream.on('data', (d) => output += d.toString());
          stream.stderr.on('data', (d) => output += d.toString());
          stream.on('close', () => {
            console.log('下载目录内容:\n' + output);

            // 检查 index.html 中的下载链接
            conn.exec('grep -n "download" ' + REMOTE_DIR + '/index.html | head -20', (err, stream) => {
              if (err) { console.log('grep 错误:', err); conn.end(); return; }
              let grepOut = '';
              stream.on('data', (d) => grepOut += d.toString());
              stream.stderr.on('data', (d) => grepOut += d.toString());
              stream.on('close', () => {
                console.log('当前下载链接:\n' + grepOut);
                console.log('\n✅ 安装包上传完成!');
                console.log('请确保网站中下载链接指向: /download/TriGenClaw-1.0.0-win-Setup.exe');
                conn.end();
              });
            });
          });
        });
      });

      writeStream.on('error', (err) => {
        console.error('上传错误:', err.message);
        conn.end();
      });

      // 显示进度
      let uploadedBytes = 0;
      readStream.on('data', (chunk) => {
        uploadedBytes += chunk.length;
        const percent = (uploadedBytes / fs.statSync(INSTALLER).size * 100).toFixed(1);
        process.stdout.write('\r上传进度: ' + percent + '% (' + (uploadedBytes / 1024 / 1024).toFixed(1) + 'MB)');
      });

      readStream.pipe(writeStream);
    });
  });
});

conn.on('error', (err) => {
  console.error('连接错误:', err.message);
});

const connectConfig = {
  host: HOST,
  port: 22,
  username: USER,
  readyTimeout: 30000,
  password: PASS,
  tryKeyboard: true,
  // Explicitly don't use key
  privateKey: undefined,
  publicKey: undefined,
};

// Remove the KEY_FILE and PASS lines at the top since we don't need them
// KEY_FILE is not used

conn.connect(connectConfig);
